components/kyma-environment-broker/internal/model.go

```diff
@@ -35,13 +36,109 @@ const (
 )
 
 type CLSInstance struct {
-	Version                  int
-	ID                       string
-	GlobalAccountID          string
-	Region                   string
-	CreatedAt                time.Time
-	ReferencedSKRInstanceIDs []string
-	RemovedBySKRInstanceID   string
+	version         int
+	id              string
+	globalAccountID string
+	region          string
+	createdAt       time.Time
+	references      []string
+	removedBy       string
+
+	events []interface{}
+}
+type CLSInstanceReferencedEvent struct {
+	SKRInstanceID string
+}
+
+type CLSInstanceUnreferencedEvent struct {
+	SKRInstanceID string
+}
+
+func NewCLSInstance(version int, id, globalAccountID, region string, createdAt time.Time, references []string, removedBy string) *CLSInstance {
```

Comment:
We need a constructor which could be used to create a "fresh" (completely new) CLSInstance. So the only arguments list is: id (it can be also generated inside the constructor), GAID, region, SKRInstanceID. All other fields you can fill, let's say:
```
func NewCLSInstance(globalAccountID string, region string, skrInstanceID string) CLSInstance {
return CLSInstance{
  version: 0,
  id: uuid.New(),
  globalAccountID: globalAccountID,
region: region,
createdAt: time.New(),
references: []string{skrInstanceId},
removedBy: "",
}
}

```

Reply:
I know the storage needs another constructor - you can have the second one (find a good name)

Reply:
Agree, I refactored it completely to use function options, which seems to be a good fit here.